File: day02/py/part2.py
#Read data 
text_file = open("../input.txt", "r")
lines = text_file.readlines()

#lines=["2x3x4","1x1x10"]

mytotal=0
for line in lines:
	thisdim=list(map(int,line.strip("\n").split("x")))
	firstdim=2*thisdim[0] + 2*thisdim[1]
	seconddim=2*thisdim[1] + 2*thisdim[2]
	thirddim=2*thisdim[0] + 2*thisdim[2]
	bow=thisdim[0]*thisdim[1]*thisdim[2]
	
	thisone=(min(firstdim,seconddim,thirddim)+bow)
	#print(firstdim,seconddim,thirddim,bow,thisone)
	mytotal+=thisone
	
print("Final:",mytotal)

File: day02/js/part2.js
//Read Data
fs  = require("fs");
array = fs.readFileSync("../input.txt").toString().split('\n');
//console.log(array);

mytotal=0
for (i = 0; i < array.length-1; i++) {//for some reason an extra blank element is being picked up
	thisdim=array[i].split("x");
	//console.log(thisdim)
	
	firstdim=2*parseInt(thisdim[0])+2*parseInt(thisdim[1])
	seconddim=2*parseInt(thisdim[1])+2*parseInt(thisdim[2])
	thirddim=2*parseInt(thisdim[0])+2*parseInt(thisdim[2])
	bow=thisdim[0]*thisdim[1]*thisdim[2]
	thisone=(Math.min(firstdim,seconddim,thirddim)+bow)
	
	mytotal+=thisone
}	

console.log("Final:",mytotal)